src/drivers/X11/Fl_X11_Screen_Driver.cxx

```diff
@@ -1369,14 +1369,17 @@ static void* value_of_key_in_schema(const char **known, const char *schema, cons
 void Fl_X11_Screen_Driver::desktop_scale_factor()
 {
   if (this->current_xft_dpi == 0.) { // Try getting the Xft.dpi resource value
-    char *s = XGetDefault(fl_display, "Xft", "dpi");
+    // Always get the up to date value
+    Display *new_dpy = XOpenDisplay(0);
```

Comment:
I wonder if this would always open the "right" display. FLTK has commandline arguments that can be used to open any possible display, not only the "default" display (the one set by the DISPLAY env var). The above code might therefore open another display with different settings than the one that's currently in use.  Theoretically. Maybe.

Reply:
Fixed. Now it's using `XDisplayString()` to get the display name from `fl_display`